File: 507.py
class Solution:
    def checkPerfectNumber(self, num: int) -> bool:
        liste = []
        if num % 2 == 0:
            for i in range(1,(num//2)+1):
                if num % i == 0:
                    liste.append(i)
        else:
            return False
        counter = 0
        for i in liste:
            counter +=i
        print(liste)
        if counter == num:
            return True
        else:
            return False

File: 507.c
bool checkPerfectNumber(int num) {
    int i;
    int counter = 0;
    for (i=1; i<(num/2+1); i++) {
        if (num % i == 0) {
            counter +=i ;
        }
    }    
    printf("%d\n", counter);
    return counter == num;
}